presto-main/src/main/java/com/facebook/presto/util/Types.java

```diff
@@ -32,4 +32,15 @@ private Types() {}
                 value.getClass().getName());
         return target.cast(value);
     }
+
+    /**
+     * Uses 64 bit hash and compact its entropy in 32 bits.
+     * @deprecated This function should only be used for keeping backward compatibility with external libraries using 32 hashes.
+     * @param hash - the 64bit (long) hash
+     * @return 32bit (int) hash created based on long hash
+     */
+    public static int longHashToIntegerHash(long hash)
+    {
+        return (int) ((hash >>> 32) ^ hash);
```

Comment:
We should use `Long.hashCode(value)` instead of this method everywhere (including the generated bytecode).